lib/adagrams.rb

```diff
@@ -0,0 +1,153 @@
+require 'csv'
+
+def draw_letters
+  all_letters_hash = {
+    "A" => 9,
+    "B" => 2,
+    "C" => 2,
+    "D" => 4,
+    "E" => 12,
+    "F" => 2,
+    "G" => 3,
+    "H" => 2,
+    "I" => 9,
+    "J" => 1,
+    "K" => 1,
+    "L" => 4,
+    "M" => 2,
+    "N" => 6,
+    "O" => 8,
+    "P" => 2,
+    "Q" => 1,
+    "R" => 6,
+    "S" => 4,
+    "T" => 6,
+    "U" => 4,
+    "V" => 2,
+    "W" => 2,
+    "X" => 1,
+    "Y" => 2,
+    "Z" => 1,
+  }
+  all_letters_array = Array.new
+  all_letters_hash.each do |letter, number|
+    number.times do
+      all_letters_array << letter
+    end
+  end
+  letters_in_hand = all_letters_array.sample(10)
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  input = input.chars
+  # Make new hash to store letters and occurrence
+  letter_count = Hash.new(0)
+  letters_in_hand.each do |letter|
+    letter_count[letter] += 1
+  end
+
+  # Check the hand array against the available letters hash
+  input.each do |letter|
+    if letters_in_hand.include?(letter)
+      letter_count[letter] -= 1
+      if letter_count[letter] < 0
+        return false
+        break
+      end
+    else
+      return false
+      break
+    end
+  end
+  return true
+end
+
+def score_word(word)
+  # Make new hash to store letters and point values
+  points = {
```

Comment:
Similar to `all_letter_hash`, it would make sense to define this hash outside the method as a global variable using all caps.